### PR TITLE
Cleanup platform and Linux distribution detection logic

### PR DESCRIPTION
**Pull request by DustinCampbell**

### CODE REVIEW

**Comment by DustinCampbell** [comment]
cc @gregg-miskelly

**Comment by DustinCampbell** [comment]
cc @rajkumar42, @gregg-miskelly, @caslan

File: src/platform.ts [hunk 1]
```diff
@@ -20,96 +95,126 @@ export enum Platform {
...
+                distributionPromise = Promise.resolve(null);
+                break;
 
-            ret.set(key, value);
+            case 'linux':
+                operatingSystem = OperatingSystem.Linux; 
+                architecturePromise = PlatformInformation.GetUnixArchitecture();
+                distributionPromise = LinuxDistribution.GetCurrent();
+                break;
         }
```
Comment (gregg-miskelly) [inline src/platform.ts]: Should we do something if it is none of these?

Comment (DustinCampbell) [inline src/platform.ts]: Yes, we should probably throw. The only other possible values are `aix`, `freebsd`, `openbsd` or `sunos` , which VS Code doesn't really support either.

**Pull request merged**